@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -398,48 +437,84 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
                 text = String(text);
 
                 if (this.m_skipShortLabels) {
-                    // Filter the lines, keep only those that are long enough for labelling.
+                    // Filter the lines, keep only those that are long enough for labelling. Also,
+                    // split jagged label paths to keep processing and rendering only those that
+                    // have no sharp corners, which would not be rendered anyway.
                     validLines = [];
 
                     const metersPerPixel = tileSizeInMeters / this.m_decodeInfo.tileSizeOnScreen;
-                    const minTileSpace =
+                    const minEstimatedLabelLength =
                         MIN_AVERAGE_CHAR_WIDTH *
                         text.length *
                         metersPerPixel *
                         SIZE_ESTIMATION_FACTOR;
+                    const minEstimatedLabelLengthSqr =
+                        minEstimatedLabelLength * minEstimatedLabelLength;
+
+                    // Work on a copy of the path.
+                    const pathsToCheck = lines.slice();
+
+                    while (pathsToCheck.length > 0) {
+                        const path = pathsToCheck.pop();
+
+                        if (path === undefined || path.length < 6) {
+                            continue;
+                        }
 
-                    // Estimate if the line is long enough for the label, otherwise ignore it for
-                    // rendering text. First, compute the bounding box in world coordinates.
-                    for (const aLine of lines) {
-                        let minX = Number.MAX_SAFE_INTEGER;
-                        let maxX = Number.MIN_SAFE_INTEGER;
-                        let minY = Number.MAX_SAFE_INTEGER;
-                        let maxY = Number.MIN_SAFE_INTEGER;
-                        for (let i = 0; i < aLine.length; i += 3) {
-                            const x = aLine[i];
-                            const y = aLine[i + 1];
-                            if (x < minX) {
-                                minX = x;
+                        let splitIndex = -1;
+
+                        for (let i = 0; i < path.length - 3; i += 3) {
```

Comment:
please add a test for this function, it should not be too difficult (simple input / output)

Reply:
added test